fftools/ffmpeg.c

```diff
@@ -1850,24 +1878,48 @@ static void print_report(int is_last_report, int64_t timer_start, int64_t cur_ti
             if(ost->need_transcode_reset) {
                 ost->need_transcode_reset = 0;
                 transcode_reset_time = cur_time;
-                transcode_reset_frame_number = frame_number;
-                transcode_reset_total = transcode_total;
-                transcode_max = transcode_us_last;
+                enc_reset_frame_number = frame_number;
+                dec_reset_frame_number = frames_decoded;
+                enc_reset_total = enc_total;
+                dec_reset_total = dec_total;
+                enc_max = enc_us_last;
+                dec_max = dec_us_last;
+                tr = 0;
             }
             if(tr > 1) {
-                int frames_since_reset = frame_number - transcode_reset_frame_number;
-                cur_transcode_pf = frames_since_reset > 0 ? (transcode_total - transcode_reset_total) / frames_since_reset / 1000.0 : transcode_us_last / 1000.0;
-                cur_transcode_max = transcode_max;
-                prev_transcode_pf = cur_transcode_pf;
-                prev_transcode_max = cur_transcode_max;
+                double total_us = transcode_reset_time ? (cur_time - transcode_reset_time) : t*1000000.0;
+                int enc_frames_since_reset = frame_number - enc_reset_frame_number;
+                int dec_frames_since_reset = frames_decoded - dec_reset_frame_number;
+                cur_enc_pf = enc_frames_since_reset > 0 ? (enc_total - enc_reset_total) / enc_frames_since_reset / 1000.0 : enc_us_last / 1000.0;
+                cur_dec_pf = dec_frames_since_reset > 0 ? (dec_total - dec_reset_total) / dec_frames_since_reset / 1000.0 : dec_us_last / 1000.0;
+                cur_enc_max = enc_max;
+                cur_dec_max = dec_max;
+                cur_enc_pct = (enc_total - enc_reset_total) / total_us;
+                cur_dec_pct = (dec_total - dec_reset_total) / total_us;
+                prev_enc_pf = cur_enc_pf;
+                prev_dec_pf = cur_dec_pf;
+                prev_enc_max = cur_enc_max;
+                prev_dec_max = cur_dec_max;
+                prev_enc_pct = cur_enc_pct;
+                prev_dec_pct = cur_dec_pct;
+#if 0                
+                if(cur_fps<10) {
```

Comment:
doesn't the format string expect a value?

Reply:
yep.. fixed